lib/tapioca/compilers/requires_compiler.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+# typed: true
+
+require_relative '../sorbet_config_parser'
+
+module Tapioca
+  module Compilers
+    class RequiresCompiler
+      extend T::Sig
+
+      sig { params(sorbet_path: String).void }
+      def initialize(sorbet_path)
+        @sorbet_path = sorbet_path
+      end
+
+      sig { returns(String) }
+      def compile
+        config = SorbetConfig.parse_file(@sorbet_path)
+        files = collect_files(config)
+        files.flat_map do |file|
+          collect_requires(file).reject do |req|
+            name = collect_required_name(T.cast(req, String))
+            next unless name
+            name_in_project?(files, T.must(name[1]))
+          end
+        end.sort.uniq.join
+      end
+
+      private
+
+      sig { params(config: SorbetConfig).returns(T::Array[String]) }
+      def collect_files(config)
+        config.paths.flat_map do |path|
+          path = (Pathname.new(@sorbet_path) / "../.." / path).cleanpath
+          if path.directory?
+            Dir.glob("#{path}/**/*.rb", File::FNM_EXTGLOB).reject do |file|
+              file_ignored_by_sorbet?(config, file)
+            end
+          else
+            [path.to_s]
+          end
+        end.sort.uniq
+      end
+
+      sig { params(file_path: String).returns(T::Enumerable[String]) }
+      def collect_requires(file_path)
+        File.read(file_path).lines.select do |line|
+          line =~ /^require ?\(?['"]/
+        end
+      end
+
+      sig { params(required_string: String).returns(T.nilable(MatchData)) }
+      def collect_required_name(required_string)
+        /['"]([^'"]+)['"]/.match(required_string)
```

Comment:
If you decide to keep this method, we can use the block arg of `match` to return a `T.nilable(String)` from this method:
```suggestion
        /['"]([^'"]+)['"]/.match(required_string) do |m| 
        	m[0]
        end
```
since the return value of `#match`, if supplied a block, is the return value of the block. And the block is only called if there was a match.